gapic/src/main/com/google/gapid/perfetto/views/BatterySummaryPanel.java

```diff
@@ -187,7 +190,12 @@ protected Hover onTrackMouseMove(TextMeasurer m, double x, double y, int mods) {
       @Override
       public Area getRedraw() {
         double redrawW = CURSOR_SIZE + HOVER_MARGIN + hovered.allSize.w + 3 * HOVER_PADDING + LEGEND_SIZE;
-        double redrawX = Math.min(mouseXpos - CURSOR_SIZE, state.getWidth() - redrawW);
+        double redrawX = mouseXpos - CURSOR_SIZE;
+        if (redrawX >= state.getWidth() - redrawW) {
+          redrawX = mouseXpos - redrawW;
+          // The cursor is now on the right side of the hover card, hence extend the redraw width.
+          redrawW += CURSOR_SIZE;
```

Comment:
This panel doesn't have the logic of doubling up the redrawW, but somehow it still works. 
I was a little bit curious so I looked into it and found the base Canvas `PanelCanvas` will handle combining together the old and new redraw areas. In this way, we could remove the logic of doubling up the redrawW in other panels.